docs/docs/recipes.md

```diff
@@ -1839,6 +1839,60 @@ Fragments can be nested inside other fragments, and multiple fragments can be us
 - [Gatsby image fragments](/docs/gatsby-image/#image-query-fragments)
 - [Example repo with co-located data](https://github.com/gatsbyjs/gatsby/tree/master/examples/gatsbygram)
 
+### Querying data client-side with `fetch`
+
+Data doesn't only have to be queried at build time and remain solely static. You can query data at runtime the same way you can fetch data in a regular React app.
+
+#### Prerequisites
+
+- A JavaScript file like a page component: `index.js`
+
+#### Directions
+
+1. In a file with a React component defined, like a page in `src/pages` or a layout component, import React hooks for `useState` and `useEffect`.
+
+```jsx:title=src/pages/index.js
+import React, { useState, useEffect } from "react"
+```
+
+2. Inside the component, wrap a function to fetch data in a `useEffect` hook so it will asynchronously retrieve data when the component mounts in the browser client. Then, `await` the result with the `fetch` API, and call the set function from the `useState` hook (in this case `setStarsCount`) to save the state variable (`starCount`) to the data returned from `fetch`.
```

Comment:
```suggestion
2. Inside the component, wrap a function to fetch data in a `useEffect` hook so it will asynchronously retrieve data when the component mounts in the browser client. Then, `await` the result with the `fetch` API, and call the set function from the `useState` hook (in this case `setStarsCount`) to save the state variable (`starsCount`) to the data returned from `fetch`.
```